### ISSUE TITLE
MA0016 is reported for Dictionary<,> parameters

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#560]
### Version of the Meziantou.Analyzer NuGet package

2.0.66

### Rule Identifier

MA0016

### Target Framework

net8.0

### C# Language version

C# 12

### Description

MA0016 is reported for parameters of type Dictionary<,> 

### Reproduction Steps

Minimal code:

```c#
[ApiController]
[Route("/api/v{version:apiVersion}/messenger/chats")]
public sealed class MessengerChatController(IMessenger messenger) : ControllerBase
{
    [HttpDelete("messages")]
    public async Task<FunctionalResult<long, Error>> DeleteMessagesAsync(
        [FromQuery(Name = "range")] Dictionary<InternalChatDescriptor, Range<long>> ranges) // diagnostic is reported here
    {
    }
}
```

**Comment by meziantou** [comment]
This is the expected behavior, but the diagnostic message and documentation must be updated.
